ADAL/src/ADDefaultEvent.m

```diff
@@ -0,0 +1,153 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "ADDefaultEvent.h"
+#import "ADEventInterface.h"
+
+@implementation ADDefaultEvent
+
+@synthesize propertyMap = _propertyMap;
+
+- (id)init
+{
+    //Ensure that the appropriate init function is called. This will cause the runtime to throw.
+    [super doesNotRecognizeSelector:_cmd];
+    return nil;
+}
+
+- (id)initWithName:(NSString*)eventName
+{
+    if (!(self = [super init]))
+    {
+        return nil;
+    }
+    
+    _propertyMap = [[self defaultParameters] mutableCopy];
+    [_propertyMap addObject:@[@"event_name", eventName]];
+    
+    return self;
+}
+
+- (void)setProperty:(NSString*)name value:(NSString*)value
+{
+    if ([NSString adIsStringNilOrBlank:name] || [NSString adIsStringNilOrBlank:value])
+    {
+        return;
+    }
+    
+    [_propertyMap addObject:@[name, value]];
+}
+
+- (NSArray*)getProperties
+{
+    return _propertyMap;
+}
+
+- (void)setStartTime:(NSDate*)time
+{
+    if (!time)
+    {
+        return;
+    }
+    
+    [_propertyMap addObject:@[@"start_time", [self getStringFromDate:time]]];
+}
+
+- (void)setStopTime:(NSDate*)time
+{
+    if (!time)
+    {
+        return;
+    }
+    
+    [_propertyMap addObject:@[@"stop_time", [self getStringFromDate:time]]];
+}
+
+- (NSString*)getStringFromDate:(NSDate*)date
+{
+    static NSDateFormatter* s_dateFormatter = nil;
+    static dispatch_once_t s_dateOnce;
+    
+    dispatch_once(&s_dateOnce, ^{
+        s_dateFormatter = [[NSDateFormatter alloc] init];
+        [s_dateFormatter setTimeZone:[NSTimeZone timeZoneWithName:@"UTC"]];
+        [s_dateFormatter setDateFormat:@"yyyy-MM-dd HH:mm:ss.SSSS"];
+    });
+    
+    return [s_dateFormatter stringFromDate:date];
+}
+
+#define SET_IF_NOT_NIL(DICT, NAME, OBJECT) \
+{ \
+if (OBJECT) \
+{ \
+[(DICT) addObject:@[(NAME), (OBJECT)]]; \
+} \
+}
+
+- (NSArray*)defaultParameters
+{
+    static NSMutableArray* s_defaultParameters = nil;
+    static dispatch_once_t s_parametersOnce;
+    
+    dispatch_once(&s_parametersOnce, ^{
+        
+        s_defaultParameters = [NSMutableArray new];
+        
+#if TARGET_OS_IPHONE
+        //iOS:
+        NSString* deviceId = [[[UIDevice currentDevice] identifierForVendor] UUIDString];
+        NSString* deviceName = [[UIDevice currentDevice] name];
+        SET_IF_NOT_NIL(s_defaultParameters, @"device_id", [[deviceId dataUsingEncoding:NSUTF8StringEncoding] base64EncodedStringWithOptions:0]);
```

Comment:
Currently it is using the device's UUID, but seems Mac does not have that. Maybe your solution yesterday should work for both iOS and Mac OS :)